### PR TITLE
Lt 1658 parties refactor

### PR DESCRIPTION
**Pull request by a-bek**
- Squash communication for parties between api and exporter.
- Post submit party document delete disabled

### CODE REVIEW

File: applications/helpers/check_your_answers.py [hunk 1]
```diff
@@ -166,10 +167,11 @@ def convert_consignee(consignee, application_id, editable):
         return {}
 
     if consignee["document"]:
-        document = _convert_document(consignee["document"], "consignee", application_id, editable)
+        document = _convert_document(consignee, "consignee", application_id, editable)
     else:
+        kwargs = {"pk": application_id, "obj_pk": consignee["id"]}
         document = convert_to_link(
-            reverse_lazy("applications:consignee_attach_document", kwargs={"pk": application_id}), "Attach document"
+            reverse_lazy("applications:consignee_attach_document", kwargs=kwargs), "Attach document"
```
Comment (andytaylorkainos) [inline applications/helpers/check_your_answers.py]: should "Attach document" not be in lite_content

File: applications/helpers/check_your_answers.py [hunk 2]
```diff
@@ -261,11 +265,15 @@ def _convert_document(document, document_type, application_id, editable):
 
     if editable:
         return convert_to_link(
-            f"/applications/{application_id}/{document_type}/document/download", "Download", include_br=True
-        ) + convert_to_link(f"/applications/{application_id}/{document_type}/document/delete", "Delete")
+            f"/applications/{application_id}/{document_type}/{party['id']}/document/download",
+            "Download",
+            include_br=True,
+        ) + convert_to_link(f"/applications/{application_id}/{document_type}/{party['id']}/document/delete", "Delete")
```
Comment (andytaylorkainos) [inline applications/helpers/check_your_answers.py]: same for "Delete" and "Download"

File: applications/views/documents.py [hunk 3]
```diff
@@ -71,13 +73,14 @@ def post(self, request, **kwargs):
         action = document_switch(request.path)["attach"]
         if len(signature(action).parameters) == 3:
             _, status_code = action(request, draft_id, data)
+            if status_code == 201:
```
Comment (andytaylorkainos) [inline applications/views/documents.py]: please use the ` HTTPStatus` library

File: applications/services.py [hunk 4]
```diff
@@ -114,132 +108,51 @@ def post_application_countries(request, pk, json):
...
-    data = post(request, APPLICATIONS_URL + str(pk) + END_USER_URL, json)
-    return data.json(), data.status_code
-
-
-def validate_end_user(request, pk, json):
+# Parties
+def validate_party(request, pk, json):
     json = add_validate_only_to_data(json)
-    data = post(request, APPLICATIONS_URL + str(pk) + END_USER_URL, json)
+    data = post(request, APPLICATIONS_URL + str(pk) + "/", json)
```
Comment (Mark-Spiers-kainos) [inline applications/services.py]: should this validate not hit again parties url? It looks like it is posting against the application just

**Pull request merged**